File: algorithms/interview-preparation-kit-2/php/code-response/spec/HackerrankInString/HackerrankInString.spec.php
<?php

use HackerrankInString\HackerrankInString;

describe('Hackerrank In String', function () {
  it('hackerrank exist in hereiamstackerrank', function () {
    expect(HackerrankInString::check('hereiamstackerrank'))->toEqual('YES');
  });

  it('hackerrank exist in hackerworld', function () {
    expect(HackerrankInString::check('hackerworld'))->toEqual('NO');
  });

  it('hackerrank exist in hhaacckkekraraannk', function () {
    expect(HackerrankInString::check('hhaacckkekraraannk'))->toEqual('YES');
  });

  it('hackerrank exist in rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt', function () {
    expect(HackerrankInString::check('rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt'))->toEqual('NO');
  });
});

File: algorithms/interview-preparation-kit-2/ruby/code-response/spec/HackerrankInString/hackerrank_in_string.spec.rb
require_relative "../../src/HackerrankInString/hackerrank_in_string"

describe "Hackerrank In String" do
  context "hackerrank exist in hereiamstackerrank" do
    it "YES" do
      expect(HackerrankInString.check('hereiamstackerrank')).to eq('YES')
    end
  end    

  context "hackerrank exist in hackerworld" do
    it "NO" do
      expect(HackerrankInString.check('hackerworld')).to eq('NO')
    end
  end    

  context "hackerrank exist in hhaacckkekraraannk" do
    it "YES" do
      expect(HackerrankInString.check('hhaacckkekraraannk')).to eq('YES')
    end
  end    

  context "hackerrank exist in rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt" do
    it "NO" do
      expect(HackerrankInString.check('rhbaasdndfsdskgbfefdbrsdfhuyatrjtcrtyytktjjt')).to eq('NO')
    end
  end    
end
